### PR TITLE
Websocket server

### PR DESCRIPTION
**Pull request by sulphite**
This branch sets up the websocket server using Bun's API and:

- assigns a unique id to incoming connections with nanoid
- adds users to lobby on arrival
- on a "join" event, assigns player to an open Room and removes them from the lobby
- max 2 players can be in a room (tested locally)
- adds types for messages

`index.ts` was moved to the root directory while troubleshooting websocket issues. I could probably move it back though

### CODE REVIEW

File: src/Room.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { ServerWebSocket } from "bun";
+
+export class Room {
+  id: string;
+  sockets: ServerWebSocket<unknown>[];
```
Comment (adrianHards) [inline src/Room.ts]: what could we use instead of unknown?

File: src/Room.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  isFull: boolean;
+
+  constructor(id: string) {
+    this.id = id;
+    this.sockets = [];
+    this.isFull = false;
+  }
+
+  addSocket(ws: ServerWebSocket<unknown>) {
```
Comment (adrianHards) [inline src/Room.ts]: eventually we'll need to add a remove socket id for users who might refresh (assuming we don't have them log in)

File: index.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      // player makes a move
+      if(isMoveMessage(message)) {
+        console.log(message.data)
+        // handle game move
+      }
+
+      const out = `${ws.data.name}: ${msg}`;
+      console.log(out)
+
+      // if (ws.publishText("room", out) !== out.length) {
```
Comment (adrianHards) [inline index.ts]: eventually remove these and console logs

**Review by adrianHards — APPROVED** [review]

File: t
Comment (adrianHards) [inline t:1]: what is this file?

File: src/Room.ts (hunk 2 above)
Comment (adrianHards) [reply src/Room.ts]: lets use a logging library instead and as you suggested wrap logging method with a decorator, e.g. Pino

File: t
Comment (sulphite) [reply t:1]: I have deleted that now

**Pull request merged**